lib/seattleflu/id3c/cli/command/etl/redcap_det_swab_n_send.py

```diff
@@ -0,0 +1,668 @@
+    #     barcode = redcap_record['corrected_barcode']
+    # elif barcode != reentered_barcode:
+    #     raise Error # TODO
+    # elif barcode != redcap_record['return_utm_barcode']:
+    #     raise Error # TODO
+
+    # TODO in production, throw error if no barcode (or skip)
+
+    specimen = {
+        "resourceType": "Specimen",
+        "identifier": [{
+            "system": f"{INTERNAL_SYSTEM}/sample",
+            "value": barcode,
+        }],
+        "subject": {
+            "type": "Patient",
+            "identifier": patient['identifier'][0],
+        }
+    }
+
+    # TODO I believe in production all samples should have a sample process date
+    received_time = redcap_record['samp_process_date']
+    if received_time:
+        specimen["receivedTime"] = convert_to_iso(received_time, "%Y-%m-%d %H:%M")
+
+    # TODO same as above comment
+    collected_time = redcap_record['collection_date']
+    if collected_time:
+        specimen["collection"] = {
+            "collectedDateTime": convert_to_iso(collected_time, "%Y-%m-%d %H:%M")
+        }
+
+    return specimen
+
+def create_resource_questionnaire_response(redcap_record: dict, patient: dict,
+    encounter: dict) -> dict:
+    """ Returns a FHIR Specimen resource. """
+
+    def create_custom_race_key(redcap_record: dict) -> List:
+        """
+        Handles the 'race' edge case by combining "select all that apply"-type
+        responses into one list.
+        """
+        race_keys = list(filter(grab_race_keys, redcap_record))
+        race_names = list(map(lambda x: redcap_record[x], race_keys))
+        return race(race_names)
+
+    def grab_race_keys(key: str) -> Optional[Match[str]]:
+        if redcap_record[key] != '':
+            return re.match('race___[0-9]{1,3}$', key)
+        else:
+            return None
+
+    def build_questionnaire_items(question: str) -> Optional[dict]:
+        return questionnaire_item(redcap_record, question, category)
+
+    coding_questions = [
+        'race',
+    ]
+
+    boolean_questions = [
+        'ethnicity',
+        'barcode_confirm',
+        'travel_states',
+        'travel_countries',
+    ]
+
+    integer_questions = [
+        'age',
+        'age_months',
+    ]
+
+    string_questions = [
+        'education',
+        'insurance',
+        'doctor_3e8fae',
+        'how_hear_sfs',
+        'samp_process_date',
+        'house_members_d5f2d9',
+        'shelter_members',
+        'where_sick',
+        # 'antiviral_0',    TODO turn on for production
+        'acute_symptom_onset',
+        'doctor_1week',
+        # 'antiviral_1',    TODO turn on for production
+        # 'poc_behaviors',  TODO turn on for production
+    ]
+
+    question_categories = {
+        'valueCoding': coding_questions,
+        'valueBoolean': boolean_questions,
+        'valueInteger': integer_questions,
+        'valueString': string_questions,
+    }
+
+    redcap_record['race'] = create_custom_race_key(redcap_record)
```

Comment:
on second thought I'll probably rename the function invoked here to `create_custom_race_value`